util/test/sub_test.py

```diff
@@ -2022,11 +2017,27 @@ def filter_errors(output_in, pre_exec_output, execgoodfile, execlog):
                 sys.stdout.write('[Executing preexec %s.preexec]\n'%(test_filename))
                 sys.stdout.flush()
                 test_preexec = './{0}.preexec'.format(test_filename)
-                p = py3_compat.Popen([test_preexec, execname, execlog, compiler],
+                p = py3_compat.Popen([test_preexec, execname, execlog, compiler,
+                                     ' '.join(envCompopts)+ ' '+compopts,
+                                     ' '.join(globalExecopts), 
+                                     ' '.join(shlex.split(execopts))],
                                      stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
                 sys.stdout.write(p.communicate()[0])
                 sys.stdout.flush()
 
+            testenv = globalTestEnv.copy()
+            # The test environment is that of this process, augmented as specified
+            if os.access(test_filename+execenvsuffix, os.R_OK):
```

Comment:
Are we only doing this for `test.execenv` files? Is there any reason not to support it for directory wide `EXECENV` files too?